### PR TITLE
Add publish Action for Nextion components

### PR DESCRIPTION
**Pull request by pkejval**
## Description:

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#7646

## Checklist:

  - [X] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/binary_sensor/nextion.rst [hunk 1]
```diff
@@ -87,6 +87,56 @@ should be prefixed with the page name (page0/page1).
...
+          state: !lambda 'return true;'
+          # These are optional. Defaults to true.
+          publish_state: true
+          send_to_nextion: true
+
+Configuration options:
+
+- **id** (**Required**, :ref:`config-id`): The ID of the Nextion switch.
+- **state** (**Required**, string, :ref:`templatable <config-templatable>`): The boolean state to publish.
+- **publish_state** (**Optional**, bool, :ref:`templatable <config-templatable>`): Publish new state to HASS. Default is true.
```
Comment (edwardtfn) [inline components/binary_sensor/nextion.rst]: I wouldn't use "HASS". I think "Publish new state to Home Assistant." makes it more clear.

File: components/binary_sensor/nextion.rst
Comment (edwardtfn) [inline components/binary_sensor/nextion.rst:1]: What if one uses `publish_state: false` and `send_to_nextion: false`? Will this function do anything?
Is there really a need for an optional `send_to_nextion`? Why not make it mandatory (always publish to Nextion) if this component is all about the Nextion?

File: components/binary_sensor/nextion.rst (hunk 1 above)
Comment (pkejval) [reply components/binary_sensor/nextion.rst]: Agreed. I copied this line from existing documentation - lambda call section.

File: components/binary_sensor/nextion.rst
Comment (pkejval) [reply components/binary_sensor/nextion.rst:1]: You are right, that call will be nonsense. But I tried to keep same functionality as lambda calls offers. You can check API: https://esphome.io/api/classesphome_1_1nextion_1_1_nextion_component_base

I can image someone needs to publish value into sensor (that triggers filters and on_value actions) and not yet update Nextion...
Yeah, it sounds overcomplicated but as I said I just tried to keep all options possible in lambdas in YAML actions.

**Pull request closed without merging**